pkg/kubernetes/scripts/virtual-machines/components/createVmDialog.jsx

```diff
@@ -98,16 +169,57 @@ class CreateVmDialog extends React.Component {
                 <label className="help-block" id='resource-text-error'>{this.state.resourceError}</label>);
         }
 
+        let name = this.state.filename ? this.state.filename : _("No files found");
+
+        let dragOverlay = null;
+        let containerClass = '';
+
+        if (this.state.isDragging) {
+            containerClass = 'disable-events';
+            dragOverlay = (
+                <div className="drag-area">
+                    <h1 className="drop-file">
+                        <b>
+                            {_("Drop file here")}
+                        </b>
+                    </h1>
+                </div>
+            );
+        }
         return (
-            <div className="modal-body modal-dialog-body-table">
-                <div className={"form-group " + resourceErrorClassName}>
+            <div className="modal-body modal-dialog-body-table drag-container"
+                 onDragEnter={this.onDragEnter}
+                 onDragLeave={this.onDragLeave}
+                 onDragEnd={this.onDragEnd}
+                 onDragOver={preventDefault}
+                 onDrop={this.onDrop}>
+                {dragOverlay}
+                <div className={containerClass}>
+                    <label className="control-label" htmlFor="file" id="file-label">
+                        {_("Browse for a local JSON file")}
+                    </label>
+                    <div className="evenly-spaced-table">
+                        <span className="evenly-spaced-cell">
+                            <input id="filename" className="form-control no-right-border" type="text"
+                                   readOnly="readonly" value={name}/>
+                        </span>
+                        <span className="evenly-spaced-table">
+                            <label className="btn btn-default" id="file-button">
+                                Browse<input id="file-input" type="file" className="hide"
+                                             onChange={this.onNewFileEvent}/>
+                            </label>
+                        </span>
+                    </div>
                     <label className="control-label required-pf" htmlFor="resource-text" id="resource-label">
-                        {_("Definition (JSON)")}
+                        {_("Or import by dragging and dropping or pasting file contents directly")}
                     </label>
+                    <div className={"form-group " + resourceErrorClassName}>
                     <textarea id="resource-text"
```

Comment:
Please fix the indentation of the textarea as well, like you already did for the error label.

Reply:
fixed